src/components/ListItem.jsx

```diff
@@ -1,22 +1,38 @@
 import { useEffect, useState } from 'react';
 import './ListItem.css';
 import { updateItem } from '../api';
-import { useStateWithStorage } from '../utils';
+import { handleNextPurchaseDate, ONE_DAY_IN_MILLISECONDS } from '../utils';
 import { increment } from 'firebase/firestore';
 
-export function ListItem({ name, itemId, purchaseTimestamp }) {
+// move current date outside of component to use it thoroughout the component
+const currentDate = new Date();
+
+export function ListItem({ item, listPath }) {
 	const [isPurchased, setIsPurchased] = useState(false);
-	const [listPath] = useStateWithStorage('tcl-shopping-list-path', null);
+	// ---------------------------------------------------------------
+	// the path from the hook is glitching in case the user has multiple empty lists:
+	// navigates to the first empty list instead of the path that was clicked on
+	// const [listPath] = useStateWithStorage('tcl-shopping-list-path', null);
+	// ---------------------------------------------------------------
+	const {
+		name,
+		id,
+		dateLastPurchased,
+		dateCreated,
+		dateNextPurchased,
+		totalPurchases,
+	} = item;
 
 	useEffect(() => {
```

Comment:
This use effect logic could be refactored as a `calculateIsPurchased` method that returns `true` or `false`. There's actually no need to have this logic live in a `useEffect` because it can be defined directly in the state initialization of `isPurchased`. Something that can be calculated as initial state directly from one of the props (`item` in this case) should not rely on a "side effect" hook (`useEffect`) to populate that state. I've added a paraphrased snippet below to illustrate the refactor I'm asking for


```
const calculateIsPurchased = // your logic her to return true/false if something has been purchased


export function ListItem({ item, listPath }) {
 	const [isPurchased, setIsPurchased] = useState(calculateIsPurchased(//input arguments from item here));
 	...
```

Reply:
i added some changes, not sure my solution works as expected. I noticed that if i uncheck an item and refresh a page, the item will still show up as checked. I would propose to add an additional 'isPurchased' field in the item document to ensure that the state of the item is recorded in the database. Not sure how to work around this!

Reply:
This might be because when we update the item, we aren't refetching the items from the database. Did you confirm that `updateItem` triggers a db refetch?